src/EnergyPlus/HVACCooledBeam.cc

```diff
@@ -567,6 +568,12 @@ namespace HVACCooledBeam {
 			OutWaterNode = CoolBeam( CBNum ).CWOutNode;
 			InitComponentNodes( 0.0, CoolBeam( CBNum ).MaxCoolWaterMassFlow, InWaterNode, OutWaterNode, CoolBeam( CBNum ).CWLoopNum, CoolBeam( CBNum ).CWLoopSideNum, CoolBeam( CBNum ).CWBranchNum, CoolBeam( CBNum ).CWCompNum );
 
+			if ( CoolBeam( CBNum ).AirLoopNum == 0 ) { // fill air loop index
+				if ( CoolBeam( CBNum ).CtrlZoneNum > 0 ) {
+					DataZoneEquipment::ZoneEquipConfig( CoolBeam( CBNum ).CtrlZoneNum ).InletNodeAirLoopNum( CoolBeam( CBNum ).AirOutNode );
```

Comment:
This is not correct (here and in other terminal units).  ZoneEquipConfig.InletNodeAirLoopNum is not indexed by true node numbers.  It's indexed by the number of inlet nodes in the ZoneEquipConfig.  All of the air terminal units should know their ADUNum and the ADUs know their AirLoopNum, so this block shouldn't be necessary.  But there may be some timing issues of when the various terminal unit ADUNums get filled and when AirDistUnit.AirLoopNum gets filled.